libazureinit/src/media.rs

```diff
@@ -396,4 +417,66 @@ mod tests {
             _ => panic!("Non-empty passwords aren't allowed"),
         };
     }
+
+    #[test]
+    fn test_wrapped_get_mount_device() {
+        // Mock function to simulate reading /etc/fstab
+        let mock_path = Path::new("/mock/mtab");
+        let mock_get_mounted_devices = || -> Result<Vec<FsEntry>, io::Error> {
```

Comment:
Instead of using mock, you could test the function by writing a valid mount table using `tempfile`. You'll want the [NamedTempFile](https://docs.rs/tempfile/latest/tempfile/struct.NamedTempFile.html) version since we pass in a `Path` to the function and so we need a name for it.

Once you make it, you can write the table to the file and pass the file path to the function to test it. Here's a (trimmed) mount table with a `udf` iso as an example, you can replace udf with iso9660 or remove it entirely to test the various function behaviors, of course:
```
sysfs /sys sysfs rw,nosuid,nodev,noexec,relatime 0 0
/dev/sda /var/tmp btrfs rw,relatime,discard=async,space_cache=v2,subvolid=5,subvol=/ 0 0
/dev/sr0 /mnt udf ro,relatime,iocharset=utf8 0 0
```

Once you write the table to the tempfile, you can call `into_temp_path()` which will close the file and give you a `tempfile::TempPath` object which will remove the file when it's dropped. It implements `AsRef<Path>` so you can call `as_ref()` on it to produce a `&Path` to give to your wrapped function.

Reply:
I am about to merge a change for this -- let me know if that matches what you were thinking. I tried to add all the scenarios I could think we might need to test, but that may have been overkill. Thank you for the review!